src/subscribeWebPushUser.ts

```diff
@@ -0,0 +1,43 @@
+// subscribeWebPushUser.ts
+import { PushSubscription } from './interfaces';
+import { Client } from './utils/client';
+
+const defaultRestAPIUrl = 'https://api.notificationapi.com';
+export const subscribeWebPushUser = (
+  applicationServerKey: string,
+  clientId: string,
+  userId: string,
+  hashUserId?: string
+): void => {
+  const client = new Client({
```

Comment:
This is another case where I don't think an abstraction is necessary. The client class or the post function looks just like a wrapper for the fetch function, but with more lines of code, without any additional functionality. This code already does everything you do with client and client.post:

```
const url = "";
const body = "";
const headers = "";
await fetch(url, {
      body: JSON.stringify(body),
      headers: headers,
      method: 'POST'
    });
```

Unless you can make the client easier and more valuable than fetch alone, future maintainers would probably just use fetch.

Reply:
I will apply your comment but I think in future it is not just `POST` method. Also, we might pass more parameters in the header.